cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -104,6 +104,13 @@ state of the cluster through the apiserver and makes changes attempting to move
 current state towards the desired state. Examples of controllers that ship with
 Kubernetes today are the replication controller, endpoints controller, namespace
 controller, and serviceaccounts controller.`,
+		PersistentPreRunE: func(*cobra.Command, []string) error {
+			// silence client-go warnings.
+			// kube-controller-manager generically watches APIs (including deprecated ones),
```

Comment:
I bet we'll want these warnings for things other than namespace lifecycle and GC.  Other than those two, I think the only reason to use deprecated APIs is to keep APIs with no future running until they're gone.  In which case we would special case them, right?

Reply:
Future use of warnings will include custom resource definition version deprecations, custom-injected warnings from admission webhooks, and "you probably didn't mean to do this" warnings on known bad values in pod specs, etc.

In all of those cases, I don't think kube-controller-manager is the right audience for those warnings, but it would be sent them anyway as it dutifully copied pod specs from deployment -> replicaset -> pod.